syntax = "proto3";

package product_info;

service ProductInfo {
    rpc addProduct(Product) returns (ProductID);
    rpc getProduct(ProductID) returns (Product);
    rpc deleteProduct(ProductID) returns (Empty);  // Используем Empty как возвращаемое значение
}

message Product {
    string id = 1;
    string name = 2;
    string description = 3;
    float price = 4;
}

message ProductID {
    string value = 1;
}

// Определение пустого сообщения Empty
message Empty {}
